test(cuda): verify roundup_power2_divisions=1 rounds to next power-of-two

Aminsed opened the pull request:
### Motivation
`PYTORCH_CUDA_ALLOC_CONF` supports `roundup_power2_divisions`; value `1` should round to the next power-of-two. Currently it’s effectively ignored and default block rounding is used.

### Root cause
`round_size()` applies power-of-two rounding only for `divisions > 1`, and the helper asserts `divisions >= 2`. Thus, `divisions == 1` bypasses PoT logic.

### Approach
Honor `divisions == 1` in `round_size()` by rounding to the next power-of-two when `size > kMinBlockSize` and the size is not already a power-of-two. Keep existing behavior for `0` and `>= 2`. No parser changes.

### Tests
Added focused subprocess CUDA test `test/test_cuda_allocator_roundup.py` covering:
- `divisions == 1`, 514 MiB → ~1 GiB reserved
- `divisions == 1`, 1 GiB exact → unchanged
- `divisions == 0` → PoT rounding disabled
- `cudaMallocAsync` backend → setting ignored

### Backward compatibility
`divisions == 0` and `>= 2` unchanged; `cudaMallocAsync` unchanged. Behavior change only for `divisions == 1` and is opt-in via env.

Aminsed committed:
cuda allocator: honor roundup_power2_divisions=1 by rounding to next power-of-two (native backend); add focused CUDA test with env-isolated subprocess

Aminsed committed:
test(cuda): add Owner(s) header to satisfy TESTOWNERS linter for allocator roundup test

Aminsed committed:
test(cuda): add run_tests() main and format per lintrunner suggestions

ngimel commented on test/test_cuda_allocator_roundup.py:
```diff
@@ -0,0 +1,100 @@
+        }
+        code, out, err = _run_subprocess_with_env(env)
+        # If CUDA not available in CI shard, accept skip via returncode 2
+        if code == 2:
```
skips are handled by skip decorators

ngimel commented on test/test_cuda_allocator_roundup.py:
```diff
@@ -0,0 +1,100 @@
+            os.path.dirname(__file__),
+            "..",
+            "tools",
+            "repro_allocator_roundup_divisions_one.py",
```
this file doesn't exist

Aminsed replied:
fixed

ngimel reviewed (commented):
You don't need to run tests in a subprocess, just do empty cache and make sure there are no large enough empty segments

Aminsed committed:
test roundup_power2_divisions=1

Add test cases for roundup_power2_divisions=1 to existing test_allocator_settings.
Tests verify that divisions=1 rounds to next power of 2, that exact powers
of 2 remain unchanged, and that divisions=0 disables rounding.

Addresses reviewer feedback by using empty_cache pattern instead of subprocess.

eqy commented on test/test_cuda.py:
```diff
@@ -4553,6 +4553,45 @@ def power2_div(size, div_factor):
             # not supported with the cudaMallocAsync backend
             self.assertEqual(pow2_div2_mem - start_mem, power2_div(nbytes_big, 2))
 
+        # test roundup_power2_divisions:1 rounds to next power of 2
```
Not an issue introduced by this PR, but for anti-flakiness we probably want the @serialTest() decorator for this test as well

Aminsed committed:
fix trailing whitespace and add @serialTest decorator

Aminsed committed:
Merge branch 'origin/main' into fix/allocator-roundup-divisions-one

Aminsed committed:
revert cutlass submodule update
